odl/trafos/wavelet.py

```diff
@@ -21,9 +21,836 @@
+            the finest level,
+        size_list[N+1] = size of original image,
+        N = number of scales
+
+    nscales : `int`
+        Number of scales in the coefficient list
+
+    Returns
+    -------
+    arr : `numpy.ndarray`
+        Flattened and concatenated coefficient array
+        The length of the array depends on the size of input image to
+        be transformed and on the chosen wavelet basis.
+        If the size of the input image is 2^n x 2^n, the lenght of the
+        wavelet coefficient array is the same.
+
+    See also
+    --------
+    pywt : http://www.pybytes.com/pywavelets/
+    """
+    size_flatCoeff = np.prod(size_list[0])
+    for kk in range(1, nscales+1):
+        size_flatCoeff = size_flatCoeff + 7*np.prod(size_list[kk])
+
+    flat_coeff = np.zeros((size_flatCoeff))
+    approx = coeff[0]
+    approx = approx.ravel()
+    stop = np.prod(size_list[0])
+    flat_coeff[0:stop] = approx
+    start = stop
+    for kk in range(1, nscales+1):
+        (aad, ada, add, daa, dad, dda, ddd) = coeff[kk]
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = aad.ravel()
+        start = stop
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = ada.ravel()
+        start = stop
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = add.ravel()
+        start = stop
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = daa.ravel()
+        start = stop
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = dad.ravel()
+        start = stop
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = dda.ravel()
+        start = stop
+        stop = start + np.prod(size_list[kk])
+        flat_coeff[start:stop] = ddd.ravel()
+        start = stop
+
+    return flat_coeff
+
+
+def array_to_pywt_coeff3d(coeff, size_list, nscales):
+    """Convert a flat array into a pywt coefficient list.
+
+    For multilevel 3D discrete wavelet transform
+
+    Parameters
+    ----------
+    coeff :  :class:`DiscreteLp.Vector`
+        A flat coefficient vector containing the approximation,
+        and detail coefficients in the following order
+        [aaaN, aadN, adaN, addN, daaN, dadN, ddaN, dddN, ...
+        aad1, ada1, add1, daa1, dad1, dda1, ddd1]
+
+    size_list : list
+       A list of coefficient sizes such that,
+
+       size_list[0] = size of approximation coefficients at the coarsest level,
+
+       size_list[1] = size of the detailed coefficients at the coarsest level,
+
+       size_list[N] = size of the detailed coefficients at the finest level,
+
+       size_list[N+1] = size of original image,
+
+       N = nscales
+
+    nscales : int
+        Number of scales in the coefficient list
+
+    Returns
+    -------
+    :attr:`coeff_list` : `list`
+        A list of coefficient organized in the following way
```

Comment:
In general I have to say that these docs are **very** good. Extremely useful to have this written down as clearly as this. Just saying.